crc32 issues on openbsd

```
==> openbsd64:     cc -pthread -fno-strict-aliasing -DNDEBUG -O2 -pipe -fPIC -O2 -pipe -fPIC -I/usr/local/include/python3.4m -c src/borg/crc32.c -o build/temp.openbsd-6.0-amd64-3.4/src/borg/crc32.o
==> openbsd64:     In file included from src/borg/_crc32/crc32.c:66,
==> openbsd64:                      from src/borg/crc32.c:433:
==> openbsd64:     src/borg/_crc32/clmul.c:36:23: error: smmintrin.h: No such file or directory
==> openbsd64:     src/borg/_crc32/clmul.c:37:23: error: wmmintrin.h: No such file or directory
==> openbsd64:     In file included from src/borg/_crc32/crc32.c:66,
==> openbsd64:                      from src/borg/crc32.c:433:
==> openbsd64:     src/borg/_crc32/clmul.c:87: error: expected ')' before '*' token
==> openbsd64:     src/borg/_crc32/clmul.c:113: error: expected ')' before '*' token
==> openbsd64:     src/borg/_crc32/clmul.c:147: error: expected ')' before '*' token
==> openbsd64:     src/borg/_crc32/clmul.c:187: error: expected ')' before '*' token
==> openbsd64:     src/borg/_crc32/clmul.c:254: error: expected declaration specifiers or '...' before '__m128i'
==> openbsd64:     src/borg/_crc32/clmul.c:254: error: expected declaration specifiers or '...' before '__m128i'
...
```

--- a/src/borg/_crc32/crc32.c
+++ b/src/borg/_crc32/crc32.c
@@ -8,6 +8,11 @@
  * target attributes or the options stack. So we disable this faster code path for clang.
  */
 #ifndef __clang__
+/*
+ * While OpenBSD uses GCC, they don't have Intel intrinsics, so we can't compile this code
+ * on OpenBSD.
+ */
+#ifndef __OpenBSD__
 #if __x86_64__
 /*
  * Because we don't want a configure script we need compiler-dependent pre-defined macros for detecting this,
@@ -59,6 +64,7 @@
 #endif
 
 #endif /* if __x86_64__ */
+#endif /* ifndef __OpenBSD__ */
 #endif /* ifndef __clang__ */
 #endif /* ifdef __GNUC__ */
 

